server/konnectors/linkedin.js

```diff
@@ -0,0 +1,512 @@
+'use strict';
+
+const request = require('request');
+const https = require('https');
+const url = require('url');
+const cheerio = require('cheerio');
+const log = require('printit')({prefix: 'Linkedin', date: true});
+
+const localization = require('../lib/localization_manager');
+const Contact = require('../models/contact');
+const Tag = require('../models/tag');
+const ContactHelper = require('../lib/contact_helper');
+const CompareContacts = require('../lib/compare_contacts');
+const linkedin = require('../lib/linkedin_helper');
+
+const ACCOUNT_TYPE = 'com.linkedin';
+
+
+module.exports = {
+  name: 'Linkedin',
+  slug: 'linkedin',
+  description: 'konnector description linkedin',
+
+  fields: {
+    login: 'text',
+    password: 'password'
+  },
+
+  models: {
+    contact: Contact
+  },
+
+  fetch: (requiredFields, callback) => {
+    log.info('Import started');
+
+    Promise.resolve(requiredFields)
+    .then(retrieveTokens)
+    .then(logIn)
+    .then(retrieveContactList)
+    .then(prepareCozyContacts)
+    .then(getOrCreateTag)
+    .then(retrieveAndSaveContacts)
+    .then(createNotificationContent)
+    .then((notifContent) => {
+      callback(null, notifContent);
+    }, (reason) => {
+      callback(reason);
+    });
+  }
+};
+
+
+/**
+* Load landing page to retrieve the csrf token needed in login request.
+* More info on csrf -> https://en.wikipedia.org/wiki/Cross-site_request_forgery
+*
+* The html parsing is done with cheerio, a "jquery like" that can be run on
+* the server side.
+*/
+function retrieveTokens(requiredFields) {
+  return new Promise((resolve, reject) => {
```

Comment:
I guess the pouchdb create a promise and return it so we can link directly to it the other promises. 
In all case when you haves a async function you need to create a promise, so in our case we have two solutions: we create ourself the promises (like this code) or we use a lib which manage himself the promise creation and return it (like your exemple).
Almost all async action are created by the request module and I juste see that he have an official promise version. So yes I probably can almost all "new Promise" terms but if you want remove all I need to move them in the Models or better in the CozyModel module.